modules/core/02-client/keeper/grpc_query_test.go

```diff
@@ -395,6 +395,106 @@ func (suite *KeeperTestSuite) TestQueryConsensusStates() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestQueryConsensusStateHeights() {
+	var (
+		req                      *types.QueryConsensusStateHeightsRequest
+		expConsensusStateHeights = []types.Height{}
+	)
+
+	testCases := []struct {
+		msg      string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"success: without pagination",
+			func() {
+				req = &types.QueryConsensusStateHeightsRequest{
+					ClientId: testClientID,
+				}
+			},
+			true,
+		},
+		{
+			"success: response contains no results",
+			func() {
+				req = &types.QueryConsensusStateHeightsRequest{
+					ClientId: testClientID,
+					Pagination: &query.PageRequest{
+						Limit:      3,
+						CountTotal: true,
+					},
+				}
+			},
+			true,
+		},
+		{
+			"success: returns concensus heights",
+			func() {
+				cs := ibctmtypes.NewConsensusState(
+					suite.consensusState.Timestamp, commitmenttypes.NewMerkleRoot([]byte("hash1")), nil,
+				)
+				cs2 := ibctmtypes.NewConsensusState(
+					suite.consensusState.Timestamp.Add(time.Second), commitmenttypes.NewMerkleRoot([]byte("hash2")), nil,
+				)
+
+				clientState := ibctmtypes.NewClientState(
+					testChainID, ibctmtypes.DefaultTrustLevel, trustingPeriod, ubdPeriod, maxClockDrift, testClientHeight, commitmenttypes.GetSDKSpecs(), ibctesting.UpgradePath, false, false,
+				)
+
+				// Use CreateClient to ensure that processedTime metadata gets stored.
+				clientId, err := suite.keeper.CreateClient(suite.ctx, clientState, cs)
+				suite.Require().NoError(err)
+				suite.keeper.SetClientConsensusState(suite.ctx, clientId, testClientHeight.Increment(), cs2)
+
```

Comment:
nit: use the testing package to fill state

```go
path := ibctesting.NewPath(suite.chainA, suite.chainB)
suite.coordinator.SetupClients(path) 

expConsensusStateHeights = append(expConsensusStateHeights, path.EndpointA.GetClientState().GetLatestHeight())

err := path.EndpointA.UpdateClient()
suite.Require().NoError(err)

expConsensusStateHeights = append(expConsensusStateHeights, path.EndpointA.GetClientState().GetLatestHeight())
```

Reply:
thanks , I'm updating my code now